CommonMark/Formatters/HtmlFormatterSlim.cs

```diff
@@ -651,6 +656,36 @@ private static void InlinesToHtml(HtmlTextWriter writer, Inline inline, CommonMa
             }
         }
 
+        private static void WriteGithubHeaderIds(HtmlTextWriter writer, Inline inline, CommonMarkSettings settings, Stack<InlineStackEntry> stack) {
+            if ((settings.AdditionalFeatures & CommonMarkAdditionalFeatures.GithubStyleHeaderIds) == 0)
+                return;
+
+            using (var tempWriter = new System.IO.StringWriter()) {
+                var tempWrapper = new HtmlTextWriter(tempWriter);
+                var tempStringBuilder = new StringBuilder();
+                InlinesToPlainText(tempWrapper, inline, stack);
+                string plaintextContent = tempWriter.ToString();
+
+                // Normalize plaintext content according to Github ID rules
+                plaintextContent = new Regex(@"[^\w\-\ ]").Replace(plaintextContent, "").Replace(" ", "-");
```

Comment:
The regexp here is the best way I can think of to emulate the behaviour of my reference Github implementation, which itself uses a (Ruby) regexp matching [^\w\-\ ] and replacing them with empty.  What would you suggest as an alternative?  That regexp works fine in .NET too.